### PR TITLE
add types

### PR DESCRIPTION
**Pull request by m8schmit**
Hi, following this issue: https://github.com/mrbungle64/ecovacs-deebot.js/issues/97
I tried with https://github.com/Dracks/ecovacs-home-desktop-clone. without success, I never understand how to use Nodemon. :thinking:.

So, I started an Angular project, with a web socket server: https://github.com/m8schmit/open-source-deebot/tree/feature/typescript-backend

After some tests, I decided to pass my backend in typescript, not to have too much conflict I had to type yours too. 

It works on my side, by defining the types in my project, but if you are interested in having them I have prepared an MR

### CODE REVIEW

File: countries.json [hunk 1]
```diff
@@ -0,0 +1,1012 @@
+{
```
Comment (m8schmit) [inline countries.json:1]: I'm not sure why it was a module. :thinking: 
With the typing it's more easy to get the value by key (using `COUNTRY_CODE`) with a json than with an array of Namespaces.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+version: '3.3'
```
Comment (m8schmit) [inline docker-compose.yml:1]: It's easier to work in Docker on my side. I can remove it if you want.

**Comment by m8schmit** [comment]
I'm not sure it will work, on my side I had to manually wrap all the index  definition into an `module`  declaration (https://github.com/m8schmit/open-source-deebot/blob/main/websocket-server/types/ecovacs-deebot/index.d.ts#L1).

File: countries.json (hunk 1 above)
Comment (mrbungle64) [reply countries.json:1]: > I'm not sure why it was a module.

I'm also not sure 😉 
This is still from the original sucks library...

> thinking With the typing it's more easy to get the value by key (using `COUNTRY_CODE`) with a json than with an array of Namespaces.

I agree that it is a good idea to change it 👍🏻

**Comment by mrbungle64** [comment]
@m8schmit 

Have you already checked it?
Do you want to create a new pull requests?

**Comment by m8schmit** [comment]
> @m8schmit
> 
> Have you already checked it? Do you want to create a new pull requests?

Yes! I'll find a clean solution and re-open a new PR.

**Pull request closed without merging**